src/ipc/CMakeLists.txt

```diff
@@ -21,3 +21,11 @@ if (CONFIG_HOST_PTABLE)
 	add_local_sources(sof
 		ipc-host-ptable.c)
 endif()
+
+# define compiler version
+if(XCC)
+	add_compile_definitions(data_structs PUBLIC XCC_TOOLS_VERSION="${XCC_TOOLS_VERSION}")
+else()
+	add_compile_definitions(data_structs PUBLIC XCC_TOOLS_VERSION="")
+endif()
```

Comment:
Can be just `add_compile_definitions(data_structs PUBLIC XCC_TOOLS_VERSION="${XCC_TOOLS_VERSION}")` as we expect toolchain script to set this variable anyway